packages/core/data-transfer/lib/encryption/__tests__/encrypt.test.ts

```diff
@@ -0,0 +1,43 @@
+import { createCipher } from '../encrypt';
+
+describe('Encryption', () => {
+  test('encrypting data with default algorithm aes-128-ecb', () => {
+    const cipher = createCipher('password');
+    const textToEncrypt = 'something ate an apple';
+    const encryptedData = cipher.update(textToEncrypt);
+
+    expect(cipher).toBeDefined();
+    expect(encryptedData).not.toBe(textToEncrypt);
```

Comment:
Since encryptedData is a `Buffer` won't this pass even if the `Buffer` still contains "something at an apple"? It might need to be something like expect(encryptedData.toString()).not.toBe(textToEncrypt).

Although one step better would be to encrypt with an alternate method (or do it again here directly with node crypto) and make sure the results match.

Reply:
Yeah, I could try the first suggestion, but I don't know about the second one, because we shouldn't really be trying to test the Crypto (we know it works) and if it doesn't there is not much we can do except replace it. but I would say I can add another test as well that compares the different algorithms to make sure they are giving different results.

And when we create the decryption we'll be able to test the whole workflow of encrypting and decrypting.

what do you think ?

Reply:
Even with the whole workflow tested, the tests are still trusting that it's encrypted without verifying, but our "encrypt" and "decrypt" functions could actually be doing (for example) rot-13 for each method and the tests would still pass.

So while it feels (and is at the moment) redundant to essentially copy/paste your node crypto calls here, it ensures that we're actually encrypting, and that we don't break it in the future (for example, maybe we switch from node crypto to some third party crypto library - the tests would still be ensuring the results of each method are the same. Or if we make some small change to the implementation that changes the encryption output, even if the whole workflow still works we'd invalidate every old export)

If it weren't a security thing, I (probably) wouldn't be so annoying about it 😆

Reply:
we can always have a second opinion @Convly ,
I personally think it's overkill to test the encryption library + we need to think about how we want to do that because the encryption will not be the same anyways, each encryption will result in a different result so there is that too.

Reply:
If I understand that correctly, @innerdvations, you're suggesting that we verify if the output of our utils matches the result of the same string encrypted directly with node crypto, right?

Reply:
Eaxctly. Otherwise we're not really testing that encryption works, only that it gives us anything other than the original input.

Reply:
I would say that it makes sense to have this verification as it would allow breaking the tests if we change how we (or a malicious PR) encrypt the data in the services.